File: TiendaApp/src/app/models/producto.ts
export class Producto {

    idProducto: number;
    marca: String;
    precio: number;
     descripcion: String;
     foto:String;

    constructor(idProducto: number, marca: String, precio:number, descripcion:String,foto:String){
        this.idProducto =  idProducto;
        this.marca = marca;
        this.precio = precio;
        this.descripcion = descripcion;
        this.foto = foto;
    }
}

File: TiendaAPI/Entities/Producto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace TiendaAPI.Entities
{
    public class Producto
    {


        [Key]
        public int idProducto { get; set; }
        public string marca { get; set; }
        public decimal precio { get; set; }
        public string descripcion { get; set; }
        public string foto { get; set; }
        [InverseProperty("Producto")]
        public virtual IEnumerable<Producto_Carro> Producto_Carros{get;set;}

    }
}
